std/datetime/date.d

```diff
@@ -3055,20 +3055,19 @@ public:
         if (isSomeString!S)
     {
         import std.algorithm.searching : countUntil;
-        import std.conv : to;
         import std.exception : enforce;
         import std.format : format;
         import std.string : strip;
 
-        immutable dstr = to!dstring(strip(isoString));
+        immutable str = strip(isoString);
```

Comment:
Does it work to use `byCodeUnit` here? Presumably, the speed-up is from getting rid of allocating the `dstring`, but you're introducing more auto-decoding in the process. If `byCodeUnit` doesn't work here at the moment (e.g. due to a function still requiring actual strings), then this is still clearly better, but I'd prefer to have `byCodeUnit` used if we can.